Homework5/SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                throw new ArgumentOutOfRangeException("Row Size can't be Zero");
+            }
+            else
+            {
+                RowIndex = rowIndex;
+            }
+
+            if (colIndex < 1)
+            {
+                throw new ArgumentOutOfRangeException("Column Size can't be Zero");
+            }
+            else
+            {
+                ColumnIndex = colIndex;
+            }
+
+            Elements = new List<(int, int, int)>();
+        }
+
+
+        public int this[int row, int column]
+        {
+            get
+            {
+
+                var tempList = Elements.Where(x => x.row == row && x.column == column).Select(x => x.value).ToList();
+                if (tempList.Count < 1)
+                {
+                    return 0;
+                }
+                else
+                {
+                    return tempList[0];
+                }
+
+            }
+            set
+            {
+                Elements.Add((row, column, value));
+
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder builder = new StringBuilder();
+            bool isNotZero = false; ;
+
+            for (int r = 0; r < RowIndex; r++)
+            {
+                builder.Append(Environment.NewLine);
+                for (int c = 0; c < ColumnIndex; c++)
+                {
+                    foreach (var item in Elements)
+                    {
+                        if (item.row == r && item.column == c)
+                        {
+                            builder.Append(item.value);
+                            isNotZero = true;
+                        }
+                    }
+                    if (!isNotZero)
+                    {
+                        builder.Append(0);
+                    }
+                    isNotZero = false;
+                }
+            }
+            return builder.ToString();
+        }
+
+        public IEnumerator<int> GetEnumerator()
+        {
+
+            bool isNotZero = false; ;
+
+            for (int r = 0; r < RowIndex; r++)
+            {
+
+                for (int c = 0; c < ColumnIndex; c++)
+                {
+                    foreach (var item in Elements)
+                    {
+                        if (item.row == r && item.column == c)
+                        {
+                            yield return item.value;
+                            isNotZero = true;
+                        }
+                    }
+                    if (!isNotZero)
+                    {
+                        yield return 0;
+                    }
+                    isNotZero = false;
+                }
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public List<(int, int, int)> GetNoZeroElements()
+        {
+
+            return Elements.OrderBy(x => x.column).ToList();
```

Comment:
The method returns both zeroes and non-zeroes. Also, they should be ordered by row